services/blockchain-app-registry/shared/metadataIndex.js

```diff
@@ -0,0 +1,158 @@
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+const BluebirdPromise = require('bluebird');
+const path = require('path');
+
+const {
+	Logger,
+	MySQL: {
+		getTableInstance,
+		getDbConnection,
+		startDbTransaction,
+		commitDbTransaction,
+		rollbackDbTransaction,
+	},
+} = require('lisk-service-framework');
+
+const { initDatabase } = require('./database/index');
+
+const applicationsIndexSchema = require('./database/schema/applications');
+const tokensIndexSchema = require('./database/schema/tokens');
+
+const { getChainIDByName } = require('./utils/chainUtils');
+const { downloadRepositoryToFS } = require('./utils/downloadRepository');
+const { getDirectories, read, getFiles } = require('./utils/fsUtils');
+
+const config = require('../config');
+
+const MYSQL_ENDPOINT = config.endpoints.mysql;
+
+const getApplicationsIndex = () => getTableInstance(
+	applicationsIndexSchema.tableName,
+	applicationsIndexSchema,
+	MYSQL_ENDPOINT,
+);
+const getTokensIndex = () => getTableInstance(
+	tokensIndexSchema.tableName,
+	tokensIndexSchema,
+	MYSQL_ENDPOINT,
+);
+
+const logger = Logger();
+
+const indexTokensInfo = async (filePath, dbTrx) => {
+	const tokensDB = await getTokensIndex();
+
+	const response = await read(filePath);
+	const tokenInfo = JSON.parse(response);
+
+	const tokenInfoToIndex = await BluebirdPromise.map(
+		tokenInfo.assets,
+		async (asset) => {
+			const result = {
+				chainID: tokenInfo.chain_id,
+				chainName: tokenInfo.chain_name,
+				description: asset.description,
+				name: asset.name,
+				Symbol: asset.symbol,
+				display: asset.display,
+				base: asset.base,
+				exponent: 8,
+				logo: JSON.stringify(asset.token_logo_URIs),
+			};
+			return result;
+		}, { concurrency: tokenInfo.assets.length },
+	);
+
+	await tokensDB.upsert(tokenInfoToIndex, dbTrx);
+};
+
+const indexChainInfo = async (filePath, dbTrx) => {
+	const applicationsDB = await getApplicationsIndex();
+
+	const response = await read(filePath);
+	const chainInfo = JSON.parse(response);
+
+	const ChainID = chainInfo.chain_id
+		? chainInfo.chain_id
+		: await getChainIDByName(chainInfo.chain_name, chainInfo.network);
+
+	const chainInfoToIndex = {
+		chainID: ChainID,
+		name: chainInfo.chain_name,
+		description: chainInfo.description,
+		title: chainInfo.title,
+		network: chainInfo.network,
+		genesisBlock: chainInfo.genesis_url,
+		homepage: chainInfo.homepage,
+		apis: JSON.stringify(chainInfo.apis),
+		images: JSON.stringify(chainInfo.images),
+		explorers: JSON.stringify(chainInfo.explorers),
+	};
+
+	await applicationsDB.upsert(chainInfoToIndex, dbTrx);
+};
+
+const indexBlockchainMetadata = async () => {
+	const directoryPath = './data';
+	const appDirPath = path.join(directoryPath, config.gitHub.appPath);
```

Comment:
Since `config.gitHub.appPath` wouldn't exist anymore, please resolve the repoName from the repoURL or simply use `app-registry`, whichever is convenient.

```suggestion
	const dataDirectory = './data';
	const appDirPath = path.join(dataDirectory, config.gitHub.appPath);
```